mongoreplay/mongo_op_handler.go

```diff
@@ -312,20 +312,37 @@ func (bidi *bidi) handleStreamStateOutOfSync(stream *stream) {
 // stream's and tries to create whole protocol messages from them.
 func (bidi *bidi) streamOps() {
 	bidi.logvf(Info, "starting")
-	for {
+	ch0 := bidi.streams[0].reassembled
+	ch1 := bidi.streams[1].reassembled
+	for ch0 != nil || ch1 != nil {
```

Comment:
actually I remember now why this is necessary. It's so that we do generate two EOFs for the two directions. This is related to the play issue.